fishtank/src/naming.ts

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Cluster } from './cluster'
+
+export function isValidName(name: string): boolean {
+  return /^[a-zA-Z0-9_-]+$/.test(name)
+}
+
+export function assertValidName(name: string): void {
+  if (!isValidName(name)) {
+    throw new Error(
+      `Invalid name: ${JSON.stringify(
+        name,
+      )}; names may only contain letters, numbers, underscores or hypens`,
+    )
+  }
+}
+
+/**
+ * Returns the name of the Docker network used for a cluster.
+ */
+export function networkName(cluster: Cluster): string {
+  assertValidName(cluster.name)
+  return cluster.name
+}
+
+/**
+ * Returns the name of the Docker container used to host a node in a cluster.
+ */
+export function containerName(cluster: Cluster, nodeName: string): string {
```

Comment:
nit, just pass the cluster.name is sufficient

Reply:
At that point the signature becomes `(string, string)`, and this can lead to subtle mistakes: e.g. one may inadvertently swap the order of the arguments. Using more specific types helps ensuring that the correct arguments are passed at compile time.

Compare:

```typescript
containerName(cluster, nodeName) // works
containerName(nodeName, cluster) // compile-time error
```

vs:

```typescript
containerName(cluster.name, nodeName) // works
containerName(nodeName, cluster.name) // works (bug!)
```

Reply:
I dont think interface with (string, string) is an anti-pattern and its something we are trying to avoid. The name of argument in the function is for user to know which value to pass in, not type. Also it would be anti-pattern if you explicitly make every argument of a function a different type. 

The reason I proposed passing cluster name for naming class because naming is a generic and independent util class. The `naming` class is not specifically for `cluster` class. As a generic util class, user should be get name without a cluster object, not just the use cases that you have a cluster object defined. Eg, if `docker` wants to use the naming function, and it does not have `cluster ` as class member currently, it will need to create a cluster object in order to use the util function.